packages/react-reconciler/src/ReactChildFiber.js

```diff
@@ -166,6 +167,16 @@ function coerceRef(
 }
 
 function throwOnInvalidObjectType(returnFiber: Fiber, newChild: Object) {
+  if (newChild.$$typeof === REACT_LEGACY_ELEMENT_TYPE) {
```

Comment:
I think this might be missing the flag check (haven't run the code internally yet). Not sure why tests wouldn't fail.

Reply:
We only get to this error handling path when there's no match, but with the flag off there would be a match.

This is just detecting which error message to provide if there's an error.